src/routes/hubspotRoute.ts

```diff
@@ -0,0 +1,51 @@
+
+router.get('/deals', isHeaderValidation(authorizationSchema), gatewayAuthentication, isAuthPlatformTokenExist('hubspot'), isQueryValidation(hubspotListSchema), hubspotController.hubspotListDeals);
+router.post('/deals', isHeaderValidation(authorizationSchema), gatewayAuthentication, isAuthPlatformTokenExist('hubspot'), isBodyValidation(hubspotCreateDealsSchema), hubspotController.hubspotCreateDeals);
+router.patch('/deals/:dealId', isHeaderValidation(authorizationSchema), gatewayAuthentication, isAuthPlatformTokenExist('hubspot'), isBodyValidation(hubspotUpdateDealsSchema), hubspotController.hubspotUpdateDeals);
+
+router.get('/analytics-views', isHeaderValidation(authorizationSchema), gatewayAuthentication, isAuthPlatformTokenExist('hubspot'), isQueryValidation(hubspotListSchema), hubspotController.hubspotListAnalyticsViews);
+router.get('/analytics-events', isHeaderValidation(authorizationSchema), gatewayAuthentication, isAuthPlatformTokenExist('hubspot'), isQueryValidation(hubspotListSchema), hubspotController.hubspotListAnalyticsEvents);
+router.get('/analytics-reports', isHeaderValidation(authorizationSchema), gatewayAuthentication, isAuthPlatformTokenExist('hubspot'), isQueryValidation(hubspotListSchema), hubspotController.hubspotListAnalyticsReports);
+
+router.get('/companies', isHeaderValidation(authorizationSchema), gatewayAuthentication, isAuthPlatformTokenExist('hubspot'), isQueryValidation(hubspotListSchema), hubspotController.hubspotListCompanies);
+router.post('/companies', isHeaderValidation(authorizationSchema), gatewayAuthentication, isAuthPlatformTokenExist('hubspot'), isBodyValidation(hubspotCreateCompanySchema), hubspotController.hubspotCreateCompanies);
+router.patch('/companies/:id', isHeaderValidation(authorizationSchema), gatewayAuthentication, isAuthPlatformTokenExist('hubspot'), isBodyValidation(hubspotUpdateCompanySchema), hubspotController.hubspotUpdateCompanies);
+router.delete('/companies/:id', isHeaderValidation(authorizationSchema), gatewayAuthentication, isAuthPlatformTokenExist('hubspot'), hubspotController.hubspotDeleteCompanies);
+
+router.get('/custom-objects', isHeaderValidation(authorizationSchema), gatewayAuthentication, isAuthPlatformTokenExist('hubspot'), isQueryValidation(hubspotListSchema), hubspotController.hubspotListCompanies);
+router.post('/custom-objects', isHeaderValidation(authorizationSchema), gatewayAuthentication, isAuthPlatformTokenExist('hubspot'), isBodyValidation(hubspotCreateCustomObjectsSchema), hubspotController.hubspotCreateCustomObjects);
+router.patch('/custom-objects/:id', isHeaderValidation(authorizationSchema), gatewayAuthentication, isAuthPlatformTokenExist('hubspot'), isBodyValidation(hubspotUpdateCustomObjectsSchema), hubspotController.hubspotUpdateCustomObjects);
+router.delete('/custom-objects/:id', isHeaderValidation(authorizationSchema), gatewayAuthentication, isAuthPlatformTokenExist('hubspot'), hubspotController.hubspotDeleteCustomObjects);
```

Comment:
This should be handled with routes for common use, for example all get methods and filters should be encapsulated in one bigger function so that we need to only access one get endpoint with params passed for specific functionality. Check this out not only for get but for rest of endpoints. Group them by functionality